### ISSUE TITLE
Replace AWS SDK with AWS SDK v2

### ISSUE THREAD

**Issue by ekristen** [ekristen/aws-nuke#187]
This is going to be a HUGE lift, however libnuke has already eased it a bit by providing context passing.

AWS SDK v1 is EOL June 31, 2025. We need to be swapped to AWS SDK v2 by then.

**Comment by Petersoj** [comment]
Yeah this will be quite the task. At least there's a decent amount of time before EOL.

**Comment by kurtmc** [comment]
@ekristen I'd be happy to help out with this if you need.

**Comment by ekristen** [comment]
Thanks! I will probably take you up on it. 

I need to convert a resource and figure out the test harness for SDKv2 so we have an example of the conversion first.

**Comment by ekristen** [comment]
Many thanks to @lucix-aws for opening #376 to bring in aws-sdk-v2 and converting `S3Bucket` and `S3Object` to using aws-sdk-v2. @lucix-aws kindly wrote tests too!

**Comment by ekristen** [comment]
@b177y I'd like to be done by then, but haven't had time to work on it, so there's a good chance I won't. All new resources have to use v2, and I'll be monitoring for any CVE or things that pop and convert immediately. I hope to find time in the coming couple of weeks to start doing bulk conversions.
